torchao/prototype/awq/example.py

```diff
@@ -206,60 +220,78 @@ def wikitext2_ppl(
         .to(device)
     )
     print(f"Time to load model: {time.time() - t0:.02f} seconds")
-    if quant.startswith("awq"):
-        quant_dtype = quant.split("-")[1]
+    if quant.startswith("awq-int4wo"):
         group_size = int(quant.split("-")[2])
-        quant_dtype = getattr(torch, quant_dtype, torch.bfloat16)
-        print(f"running {quant_dtype} calibration")
-        t0 = time.time()
-        # insert observers to find average magnitude and calculate scales
-        insert_awq_observer_(
-            model,
-            validation_size,
-            sequence_length,
-            quant_dtype=quant_dtype,
-            group_size=group_size,
-        )
-        calibration_data = get_calib_dataset(
-            tokenizer=tokenizer, n_samples=calibration_size, block_size=sequence_length
+        print(f"running {quant} quantization with group size {group_size}")
+        # TODO: this is temporary, we'll be using Int4WeightOnlyConfig soon
+        from torchao.quantization import FbgemmConfig
+
+        # use_hqq = True
+        # base_config = Int4WeightOnlyConfig(group_size=group_size, use_hqq=use_hqq)
+        base_config = FbgemmConfig(
+            input_dtype=torch.bfloat16,
+            weight_dtype=torch.int4,
+            output_dtype=torch.bfloat16,
+            block_size=[1, group_size],
+            preshuffle=False,
         )
-        for batch in calibration_data:
-            model(batch.to(device))
-            batch.to("cpu")
-        print(f"time for calibration: {time.time() - t0:.02f} seconds")
-
-        is_observed_linear = lambda m, fqn: isinstance(m, AWQObservedLinear)
-        use_hqq = "hqq" in quant
-        print(f"running {quant_dtype} quantization")
+        print(f"running {quant} prepare and calibrate")
         t0 = time.time()
-        awq_uintx_config = awq_uintx(
-            quant_dtype=quant_dtype, group_size=group_size, use_hqq=use_hqq
-        )
-        if "xpu" in device:
-            awq_uintx_config.layout = Int4XPULayout()
+        quant_config = AWQConfig(base_config, step="prepare")
+
         quantize_(
             model,
-            awq_uintx_config,
-            is_observed_linear,
+            quant_config,
         )
-        print(f"time for quantization: {time.time() - t0:.02f} seconds")
-        if model_save_path is not None:
-            print(f"Saving model to {model_save_path}")
-            torch.save(model, model_save_path)
+        from torchao._models._eval import TransformerEvalWrapper
+
+        TransformerEvalWrapper(
+            model=model.to(device),
+            tokenizer=tokenizer,
+            max_seq_length=max_seq_length,
+            device=device,
+        ).run_eval(
+            tasks=tasks,
+            limit=calibration_limit,
+        )
```

Comment:
Hi @jerryzh168 Looks like this part does not work with `PPL` on my side. It seems that we should use `wiki2_eval` for `PPL`. How was it when you ran?

Reply:
oh we can probably can remove PPL, I'm using lm-eval tasks to do evals

Reply:
OK. Perhaps we need to update the default value of `--tasks`

Reply:
Do you have a preference for the default value, i.e. the default task?

Reply:
Hi @jerryzh168 I have created a PR for CPU: https://github.com/pytorch/ao/pull/2688 Please take a look. Thanks.